qiskit/transpiler/passes/layout/sat_layout_for_commuting_gate_router.py

```diff
@@ -0,0 +1,301 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=too-many-function-args
+
+"""SATLayout pass to find a layout for commuting gates using subgraph isomorphism and a SAT solver"""
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from itertools import combinations
+from threading import Timer
+
+import networkx as nx
+import numpy as np
+from pysat.formula import CNF, IDPool
+from pysat.solvers import Solver
```

Comment:
Maybe a question for the Qiskit team but maybe we make this dependency optional?